doc/rst/technotes/libraries.rst

```diff
@@ -792,6 +792,12 @@ via the generated library.
 
 Other configurations may also become supported in the future.
 
+Host and Target Compilers
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Multilocale libraries currently require the host and target compiler to be
+the same. We expect this to change in the near future.
```

Comment:
I might be more specific and say something like "in the future, the client stub will be compiled with the host compiler and the server executable with the target compiler" (where I'm not sure whether there are other/better terms that we use in describing these features than "client stub" and "server executable").

Also, does it require them to literally be the same, or simply to be compatible?  E.g., is `gnu` as the host compiler and `cray-prgenv-gnu` as the target reasonable, or do they literally both need to be `cray-prgenv-gnu`?  If the former, I'd say "require [them] to be compatible" or something like that (where I'm intentionally being a bit vague in terms of what "compatible" means).